ts/sagas/wallet.ts

```diff
@@ -24,10 +90,203 @@ function* fetchTransactions(
+    // either the QR code has been read, or the
+    // data has been entered manually. Store the
+    // payload and proceed with showing the
+    // transaction information fetched from the
+    // pagoPA proxy
+    const {
+      rptId,
+      initialAmount
+    }: { rptId: RptId; initialAmount: AmountInEuroCents } = action.payload;
+
+    // TODO: fetch the data from the pagoPA proxy
+    const verificaResponse: PaymentRequestsGetResponse = {
+      importoSingoloVersamento: 10052 as ImportoEuroCents,
+      codiceContestoPagamento: "6793ad707f9b11e888482902221575ae" as CodiceContestoPagamento,
+      ibanAccredito: "IT17X0605502100000001234567" as Iban,
+      causaleVersamento: "IMU 2018",
+      enteBeneficiario: {
+        identificativoUnivocoBeneficiario: "123",
+        denominazioneBeneficiario: "Comune di Canicattì",
+        codiceUnitOperBeneficiario: "01",
+        denomUnitOperBeneficiario: "CDC",
+        indirizzoBeneficiario: "Via Roma",
+        civicoBeneficiario: "23",
+        capBeneficiario: "92010",
+        localitaBeneficiario: "Canicattì",
+        provinciaBeneficiario: "Agrigento",
+        nazioneBeneficiario: "IT"
+      } as EnteBeneficiario
+    };
+    yield put(
+      paymentTransactionSummaryFromRptId(rptId, initialAmount, verificaResponse)
+    );
+  } else {
+    yield put(paymentTransactionSummaryFromBanner());
+  }
+  // also, show summary screen
+  yield put(navigateTo(ROUTES.PAYMENT_TRANSACTION_SUMMARY));
+}
+
+function* continueWithPaymentMethodsHandler(
+  _: PaymentRequestContinueWithPaymentMethods
+) {
+  // find out whether a payment method has already
+  // been defined as favorite. If so, use it and
+  // ask the user to confirm it
+  // Otherwise, show a list of payment methods available
+  // TODO: if no payment method is available (or if the
+  // user chooses to do so), allow adding a new one.
+  const favoriteCard: Option<number> = yield select(getFavoriteWalletId);
+  if (favoriteCard.isSome()) {
+    // show card
+    yield put(paymentRequestConfirmPaymentMethod(favoriteCard.value));
+  } else {
+    // show list
+    yield put(paymentRequestPickPaymentMethod());
+  }
+}
+
+function* confirmPaymentMethodHandler(
+  action: PaymentRequestConfirmPaymentMethod
+) {
+  const walletId = action.payload;
+  yield put(paymentConfirmPaymentMethod(walletId));
+  yield put(navigateTo(ROUTES.PAYMENT_CONFIRM_PAYMENT_METHOD));
+}
+
+function* pickPaymentMethodHandler(_: PaymentRequestPickPaymentMethod) {
+  // show screen with list of payment methods available
+  yield put(paymentPickPaymentMethod());
+  yield put(navigateTo(ROUTES.PAYMENT_PICK_PAYMENT_METHOD));
+}
+
+function* completionHandler(_: PaymentRequestCompletion) {
+  // -> it should proceed with the required operations
+  // and terminate with the "new payment" screen
+
+  // do payment stuff (-> pagoPA REST)
+  // retrieve transaction and store it
+  // mocked data here
+  // tslint:disable-next-line saga-yield-return-type
```

Comment:
why the disabled tslint? `selectedPaymentMethodSelector` returns a `Wallet`, this shouldn't trigger an error...

Reply:
Correct -- I accidentally added it there as well. The other yields need it though (the problem being with the fact that the return value of the `select` is not immediately assigned to a variable)